test/functional/ui/output_spec.lua

```diff
@@ -22,6 +22,7 @@ describe("shell command :!", function()
 
   after_each(function()
     screen:detach()
+    session.command('call jobstop(b:terminal_job_id)')
```

Comment:
Oops, the detach allowed the process to keep running after teardown. I don't think I had a good reason for that, maybe just removing the detach would achieve the same thing.